apps/api/src/middleware/error.middleware.ts

```
@@ -0,0 +1,39 @@
+  res.status(500).json({
+    success: false,
+    message: err.message || 'Something went wrong',
+    code: 'InternalServerError'
+  });
+}
+
+/**
+ * Middleware to catch async errors in routes
+ */
+// eslint-disable-next-line @typescript-eslint/no-unsafe-function-type
+export const catchAsync = (fn: Function) => (req: Request, res: Response, next: NextFunction): void => {
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)